docs/UsersGuide/source/ContributorsGuide.rst

```diff
@@ -0,0 +1,359 @@
+
+
+
+Contributor Requirements
+===========================
+
+Preparing Code for Contribution to the UFS SRW Application
+------------------------------------------------------------
+
+All changes should be associated with a GitHub Issue. If developers are working on a change, they should search the existing issues in the appropriate repository (``ufs-srweather-app`` and/or ``regional_workflow``). If an issue does not exist for the work they are doing, they should create one prior to opening a new pull request (see :numref:`Step %s <Issue>`).
+
+**Guidelines for All Modifications:**
+    * All changes should adhere to the Code and Configuration Standards detailed in :numref:`Section %s <ContribStandards>`. 
+    * For changes to the ``scripts``, ``ush``, or ``jobs`` directories (within ``ufs-srweather-app/regional_workflow``), developers should follow the `NCO Guidelines <https://www.nco.ncep.noaa.gov/idsb/implementation_standards/>`__ for what is incorporated into each layer. 
+    * Developers should ensure that their contributions work with the most recent version of the ``ufs-srweather-app``, including all the specific up-to-date hashes of each subcomponent.
+    * Modifications should not break any existing supported capabilities on any supported platforms.
+    * Developers will not be required to run tests on *all* supported platforms, but if a failure is pointed out by another reviewer (or by automated testing), then the developer should work with reviewers and code managers to ensure that the problem is resolved prior to merging.
+    * If possible, developers should run a fundamental test suite (see :numref:`Section %s <Testing>`) on at least one supported platform and report on the outcome in the PR template.
+    * If changes are made to ``regional_workflow``, a corresponding PR to ``ufs-srweather-app`` should be opened to update the regional_workflow hash. 
+    * Update the ``.rst`` documentation files where appropriate as part of the PR. If necessary, contributors may update the documentation in a subsequent PR. In these cases, the contributor should open an issue reflecting the need for documentation (see :numref:`Step %s <Issue>`) and include the issue number and explanation in the Documentation section of the initial PR template. 
+
+**Guidelines for New Features:**
+    * Components should have a mechanism for portability and platform-independence; code that is included in the SRW App should not be tied to specific platforms. 
+    * New components should be able to build using the standard supported NCEPLIBS environment (currently `HPC-Stack <https://github.com/NOAA-EMC/hpc-stack>`__).
+    * New entries in ``Externals.cfg`` should only be repositories from “official” sources; either the `UFS Community GitHub organization <https://github.com/ufs-community>`__ or another NOAA project organization.
+    * Document how to use the new component, and include this information in the PR. 
+
+
+.. _Testing: 
+
+Testing
```

Comment:
Will the document that @mkavulich created ultimately be linked in this chapter (once it exists on the wiki or elsewhere) for more details on this? Or is it all here, just reordered? I didn't do a thorough comparison.

Reply:
I can definitely do that in an update if the SRW team wants that. I think everything that is pertinent to contributors is included here. A lot of the details from the code management doc apply only to the code managers, not to contributors (unless the contributors are also code managers, but then they probably have access to the doc anyway. ;) ). In other words, I view Mike's doc as the equivalent of a Teacher's Guide, whereas this UG chapter is more like "student instructions." The audience is a bit different. That said, couldn't hurt to include a link once the doc's ready!